tracer/src/Datadog.Trace/Ci/ITRClient.cs

```diff
@@ -0,0 +1,428 @@
+
+        var agentlessUrl = _settings.AgentlessUrl;
+        if (!string.IsNullOrWhiteSpace(agentlessUrl))
+        {
+            _searchCommitsUrl = new UriBuilder(agentlessUrl)
+            {
+                Path = "api/v2/git/repository/search_commits"
+            }.Uri;
+
+            _packFileUrl = new UriBuilder(agentlessUrl)
+            {
+                Path = "api/v2/git/repository/packfile"
+            }.Uri;
+        }
+        else
+        {
+            _searchCommitsUrl = new UriBuilder(
+                scheme: "https",
+                host: "api." + _settings.Site,
+                port: 443,
+                pathValue: "api/v2/git/repository/search_commits").Uri;
+
+            _packFileUrl = new UriBuilder(
+                scheme: "https",
+                host: "api." + _settings.Site,
+                port: 443,
+                pathValue: "api/v2/git/repository/packfile").Uri;
+        }
+    }
+
+    public async Task<long> UploadRepositoryChangesAsync()
+    {
+        Log.Debug("ITR: Uploading Repository Changes...");
+        var gitOutput = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", "log --format=%H -n 1000 --since=\"1 month ago\"", _workingDirectory)).ConfigureAwait(false);
+        var localCommits = gitOutput.Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+        if (localCommits.Length == 0)
+        {
+            return 0;
+        }
+
+        var remoteCommitsData = await SearchCommitAsync(localCommits).ConfigureAwait(false);
+        return await SendObjectsPackFileAsync(localCommits[0], remoteCommitsData).ConfigureAwait(false);
+    }
+
+    private async Task<string[]> SearchCommitAsync(string[] localCommits)
+    {
+        if (localCommits is null)
+        {
+            return null;
+        }
+
+        Log.Debug("ITR: Searching commits...");
+
+        var commitRequests = new CommitRequest[localCommits.Length];
+        for (var i = 0; i < localCommits.Length; i++)
+        {
+            commitRequests[i] = new CommitRequest(localCommits[i]);
+        }
+
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var jsonPushedSha = JsonConvert.SerializeObject(new DataArrayEnvelopeWithMeta<CommitRequest>(commitRequests, repository));
+        var jsonPushedShaBytes = Encoding.UTF8.GetBytes(jsonPushedSha);
+
+        return await WithRetries(InternalSearchCommitAsync, jsonPushedShaBytes, MaxRetries).ConfigureAwait(false);
+
+        async Task<string[]> InternalSearchCommitAsync(byte[] state, bool finalTry)
+        {
+            var request = _apiRequestFactory.Create(_searchCommitsUrl);
+            request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            Log.Debug("ITR: Searching commits from: {url}", _searchCommitsUrl.ToString());
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Comment:
must be disposed
```suggestion
            using var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3045/commits/ee3048f6cb4f6c1027a372cdedc58883b5193527